pkg/server/ca/manager/journal.go

```diff
@@ -25,61 +31,69 @@ const (
 	journalPEMType = "SPIRE CA JOURNAL"
 )
 
-type JournalEntries = journal.Entries
-type X509CAEntry = journal.X509CAEntry
-type JWTKeyEntry = journal.JWTKeyEntry
+type journalEntries = journal.Entries
+type x509CAEntry = journal.X509CAEntry
+type jwtKeyEntry = journal.JWTKeyEntry
+
+type journalConfig struct {
+	cat      catalog.Catalog
+	log      logrus.FieldLogger
+	filePath string
+}
 
 // Journal stores X509 CAs and JWT keys on disk as they are rotated by the
 // manager. The data format on disk is a PEM encoded protocol buffer.
 type Journal struct {
-	path string
+	config *journalConfig
 
-	mu      sync.RWMutex
-	entries *JournalEntries
+	mu                    sync.RWMutex
+	activeX509AuthorityID string
+	caJournalID           uint
+	entries               *journalEntries
 }
 
-func LoadJournal(path string) (*Journal, error) {
-	j := &Journal{
-		path:    path,
-		entries: new(JournalEntries),
-	}
-
-	pemBytes, err := os.ReadFile(path)
-	if err != nil {
-		if os.IsNotExist(err) {
+func LoadJournal(ctx context.Context, config *journalConfig) (*Journal, error) {
+	if fflag.IsSet(fflag.FlagCAJournalInDatastore) {
+		// Look for the CA journal of this server in the datastore.
+		j, err := loadJournalFromDS(ctx, config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load journal from datastore: %w", err)
+		}
+		if j != nil {
```

Comment:
>Since it is a feature that someone enabled intentionally I think we have 2 ways to follow here:
1- add a warning if flag is set but no journal found in DS, in a perferct world this means that it was unable to found and in next iteration warning wil be done.

I don't think that we should log a warning. This is the expected behavior for all the systems running for the first time with this fflag enabled in 1.8.x, and will be the expected behavior for all the systems running for the first time in 1.9.0.

>2- create a migration code that try to load things from disk if not found, and then fails if that is no possible, with this we can be sure that in next iteratrion we'll be sure that we'll load from ds.

That's what it does. If there is no CA record for this server, it will load it from disk. If there is a failure when trying to load from disk, we fail with an error and it will try the same the next time. Note that having no journal is also a valid scenario. No CA record, no disk file, doesn't cause an error and will start with an empty journal.
The migration logic is pretty straightforward. The mechanism consists in just writing the next update on the database and on disk, when this feature is enabled. This way, the journal is always backed on disk during the 1.9.x timeframe to allow downgrades to 1.8.x.